File: src/main/java/org/chousik/collection/builder/collectors/IntegerCollector.java
package org.chousik.collection.builder.collectors;

import org.chousik.collection.validators.IValidator;
import org.chousik.exception.ScriptExecutionError;

public class IntegerCollector extends NumberCollector<Integer>{

    @Override
    public Integer ask(String name, IValidator<Integer> validator) throws ScriptExecutionError {
        return askNumber(name, validator, Integer::parseInt);
    }
}

File: src/main/kotlin/collection/builder/collectors/IntegerCollector.kt
package org.chousik.collection.builder.collectors

import org.chousik.collection.validators.IValidator

class IntegerCollector : NumberCollector<Int?>() {

    override fun ask(name: String?, validator: IValidator<Int?>?): Int? {
        return askNumber(name!!, validator!!,  Integer::parseInt)
    }
}